custom_model_runner/datarobot_drum/resource/components/Python/prediction_server/prediction_server.py

```diff
@@ -92,6 +86,19 @@ def predict():
             self._stats_collector.disable()
             return response, response_status
 
+        @model_api.route("/predictUnstructured/", methods=["POST"])
+        def predict_unstructured():
+            logger.debug("Entering predict() endpoint")
+
+            self._stats_collector.enable()
+            self._stats_collector.mark("start")
+
+            response, response_status = self.do_predict_unstructured(logger=logger)
```

Comment:
I would organise this as a context manager or at least wrap the do_predict_unstructured into try/finally to handle stats.

```
with stats_collector():
    response, response_status = self.do_predict_unstructured(logger=logger)
```

Reply:
it is a good pattern to avoid cases when you called enable/start but didnt call finish/disable because of exception